cpp/src/text/replace.cu

```diff
@@ -125,38 +208,57 @@ std::unique_ptr<cudf::column> replace_tokens(cudf::strings_column_view const& st
   auto replacements_column = cudf::column_device_view::create(replacements.parent(), stream);
   cudf::string_view d_delimiter(delimiter.data(), delimiter.size());
   replace_tokens_fn replacer{*strings_column,
+                             d_delimiter,
                              targets_column->begin<cudf::string_view>(),
                              targets_column->end<cudf::string_view>(),
-                             *replacements_column,
-                             d_delimiter};
+                             *replacements_column};
+
+  // copy null mask from input column
+  rmm::device_buffer null_mask = copy_bitmask(strings.parent(), stream, mr);
+
+  // this utility calls replacer to build the offsets and chars columns
+  auto children = cudf::strings::detail::make_strings_children(
+    replacer, strings_count, strings.null_count(), mr, stream);
+
+  // return new strings column
+  return cudf::make_strings_column(strings_count,
+                                   std::move(children.first),
+                                   std::move(children.second),
+                                   strings.null_count(),
+                                   std::move(null_mask),
+                                   stream,
+                                   mr);
+}
+
+std::unique_ptr<cudf::column> filter_tokens(cudf::strings_column_view const& strings,
+                                            cudf::size_type min_token_length,
+                                            cudf::string_scalar const& replacement,
+                                            cudf::string_scalar const& delimiter,
+                                            cudaStream_t stream,
+                                            rmm::mr::device_memory_resource* mr)
+{
+  CUDF_EXPECTS(replacement.is_valid(), "Parameter replacement must be valid");
+  CUDF_EXPECTS(delimiter.is_valid(), "Parameter delimiter must be valid");
+
+  cudf::size_type const strings_count = strings.size();
+  if (strings_count == 0) return cudf::make_empty_column(cudf::data_type{cudf::type_id::STRING});
```

Comment:
```suggestion
  if (strings_count == 0) return cudf::empty_like(strings);
```

Reply:
So I don't really think this is better. 
The code for `cudf::empty_like()` is
https://github.com/rapidsai/cudf/blob/0f0a8dfceb479e811263227b0d8904c8bc068e75/cpp/src/copying/copy.cpp#L77-L81
And calling this would require including an additional header file: [copying.hpp](https://github.com/rapidsai/cudf/blob/branch-0.15/cpp/include/cudf/copying.hpp).
Is there some reason `cudf::empty_like()` should be used over `cudf::make_empty_column` ?

Reply:
Only because it's 38 fewer characters for the same result. If you'd rather stick with `make_empty_column`, that's fine.